_posts/2018-10-22-the-v-for-key.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: post
+title: "The Key to the Vue v-for"
+date: 2018-10-22 11:36:00
+tags:
+- JavaScript
+- Vue
+categories:
+- JavaScript
+twitter_text: "The Key to the Vue v-for"
+authors: Jaime Jones
+image: https://images.unsplash.com/photo-1522794338816-ee3a17a00ae8?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=28707e35e45dd456c20daa5dee1b5396&auto=format&fit=crop&w=1267&q=80
+image_url: https://unsplash.com/photos/o9KNLaITFYw
+image_credit: Debby Hudson
+---
+
+If you're familiar with Vue at all, you're familiar with the `v-for`. And if you've used any other front end frameworks, they each provide ways to loop over data and markup within your HTML template. The purpose of this post is to talk about the `key` attribute that is required on every `v-for`, and what makes it so important.
+
+I'll be the first to admit, when I started writing Vue, the `key` attribute wasn't required. When it become required, I shrugged, simply added an index, and moved on.
+```html
+<h1 v-for="(thing, index) in things" :key="index">{{ thing }}</h1>
+```
+
+This article really is about the `key` attribute which is now requried on all usages of `v-for` and some of the issues that you may run into if you go the common route of using the `index`. I'd recommend reading the actual [Vue documentation](https://vuejs.org/v2/guide/list.html#key) for it as either a follow-up or some background knowledge.
```

Comment:
Is it actually *required*? Is it more of a strongly suggested thing (perhaps enforced by a linter)?

Reply:
I believe it is more a strongly suggested thing by the linter, evidenced by the fact that the default is actually the index usage. I also spelled `required` wrong there.